docs/guide/search/criteria_reference/objectstateidentifier_criterion.md

```diff
@@ -0,0 +1,14 @@
+# ObjectStateIdentifier Criterion
+
+The [`ObjectStateIdentifier` Search Criterion](https://github.com/ezsystems/ezpublish-kernel/blob/v8.0.0-beta3/eZ/Publish/API/Repository/Values/Content/Query/Criterion/ObjectStateId.php)
+searches for content based on its Object State identifier.
+
+## Arguments
+
+- `value` - int(s) representing the Object State identifier(s)
+
+## Example
+
+``` php
+$query->query = new Criterion\ObjectStateIdentifier(['ready']);
```

Comment:
It's also possible to pass second argument which is object state group:

```php
$query->query = new Criterion\ObjectStateIdentifier('not_locked', 'ez_lock');
```

I think we can have multiple examples to show this argument is optional.

Also first argument can be a string or an array of strings.